shm/CMakeLists.txt

```
@@ -0,0 +1,64 @@
+cmake_minimum_required(VERSION 2.8.2)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Download and unpack googletest at configure time
+configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
+execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+  RESULT_VARIABLE result
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
+if(result)
+  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
+endif()
+execute_process(COMMAND ${CMAKE_COMMAND} --build .
+  RESULT_VARIABLE result
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
+if(result)
+  message(FATAL_ERROR "Build step for googletest failed: ${result}")
+endif()
+
+# Prevent overriding the parent project's compiler/linker
+# settings on Windows
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+
+# Add googletest directly to our build. This defines
+# the gtest and gtest_main targets.
+add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src
+                 ${CMAKE_CURRENT_BINARY_DIR}/googletest-build
+                 EXCLUDE_FROM_ALL)
+
+# The gtest/gtest_main targets carry header search path
+# dependencies automatically when using CMake 2.8.11 or
+# later. Otherwise we have to add them here ourselves.
+if (CMAKE_VERSION VERSION_LESS 2.8.11)
+  include_directories("${gtest_SOURCE_DIR}/include")
+endif()
+
+project(TradeSimulator)
+enable_testing()
+set(SOURCE
+    src/alcohol.cpp
+    src/cargo.cpp
+    src/coordinates.cpp
+    src/fruit.cpp
+    src/island.cpp
+    src/item.cpp
+    src/map.cpp
+    src/player.cpp
+    src/ship.cpp
+    src/store.cpp
+)
+include_directories(include)
+add_library(lib STATIC ${SOURCE})
+
+add_executable(${PROJECT_NAME} src/main.cpp)
+add_executable(${PROJECT_NAME}-ut tests/test.cpp)
+
+add_compile_options(${PROJECT_NAME} -Wall -Wextra -Wconversion -pedantic -Werror)
+
+target_link_libraries(${PROJECT_NAME}-ut gtest_main)
+target_link_libraries(${PROJECT_NAME}-ut lib)
```

You can merge those two:
```suggestion
target_link_libraries(${PROJECT_NAME}-ut lib gtest_main)
```